### PR TITLE
prevent fetching to many orcids from cristin api

### PR DESCRIPTION
**Pull request by sondrev**
If we have a publication with too many contributors we should skip looking them up though Cristin. 
https://unit.atlassian.net/browse/NP-46635

### CODE REVIEW

File: create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/service/IdentityUpdater.java [hunk 1]
```diff
@@ -46,24 +47,30 @@ private static Optional<List<Contributor>> extractPossibleContributors(Publicati
...
+            return;
+        }
+
+        contributorsWithOnlyOrcid
+            .forEach(contributor -> updateIdentifierIfFoundFromOrcid(cristinProxyClient, contributor.getIdentity())
+        );
     }
 
-    private static boolean hasNoIdentifierButCanPossiblyBeFetchedUsingOrcid(Identity identity) {
+    private static boolean hasOcidButNotIdentifier(Contributor contributor) {
```
Comment (torarnet) [inline create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/service/IdentityUpdater.java:72]: O(r)cid

File: create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java [hunk 2]
```diff
@@ -89,13 +94,27 @@ public void shouldNotUpdateIdentifierWhenIdentifierAlreadyExistsWhenEnrichingPub
...
+        var publication = createPublicationWithIdentites(List.of());
         var cristinProxyClient = mock(CristinProxyClient.class);
         var updatedPublication = IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
 
         assertThat(updatedPublication, equalTo(publication));
     }
 
+    @Test
+    public void shouldNotEnrichContributorsWhenMoreThan10AreMissingCristinIdentifier() {
+        var identies = Collections.nCopies(11, 1).stream().map(i -> createIdentityWithOrcid()).toList();
```
Comment (torarnet) [inline create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java:106]: identities

**Review by torarnet — APPROVED** [review]

File: create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java [hunk 3]
```diff
@@ -89,13 +94,27 @@ public void shouldNotUpdateIdentifierWhenIdentifierAlreadyExistsWhenEnrichingPub
...
+        var publication = createPublicationWithIdentites(identies);
```
Comment (brinxmat) [inline create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java:107]: Identities

File: create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java [hunk 4]
```diff
@@ -89,13 +94,27 @@ public void shouldNotUpdateIdentifierWhenIdentifierAlreadyExistsWhenEnrichingPub
...
-        var publication = createPublicationWithIdentity(null);
+        var publication = createPublicationWithIdentites(List.of());
         var cristinProxyClient = mock(CristinProxyClient.class);
         var updatedPublication = IdentityUpdater.enrichPublicationCreators(cristinProxyClient, publication);
 
         assertThat(updatedPublication, equalTo(publication));
     }
 
+    @Test
+    public void shouldNotEnrichContributorsWhenMoreThan10AreMissingCristinIdentifier() {
```
Comment (brinxmat) [inline create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/service/IdentityUpdaterTest.java:105]: I doubt that Cristin identifier will ever be populated in any case, so I'd maybe change the name here to `shouldNotEnrichContributorsWhenMoreThanTenUnverifiedContributorsExist` 

"Unverified" is the term for those authors that lack a Cristin ID.

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Pull request merged**